### DISCUSSION TITLE
From RGB to grayscale

### DISCUSSION THREAD

**Discussion by jumpjack** [ImageMagick/ImageMagick#5407] [Digital Image Processing]
![image](https://user-images.githubusercontent.com/1620953/182612299-39e23141-e0e2-4945-ae9a-3e70744a3e5a.png)

https://mars.nasa.gov/resources/8658/topographic-map-of-pathfinder-landing-site/

Is it possibile to turn this image from RGB gradient to grayscale depthmap? A simple grayscale conversion in any editor does not do the trick, because both red and blue areas appear dark, while one should be turned into black and the other into white.

**Comment by fmw42** [comment]
What are you going to do with the black contour lines when mapping to depth?

**Reply by fmw42** [reply]
Map the Hue channel values to depth not the RGB to grayscale image

**Reply by jumpjack** [reply]
> What are you going to do with the black contour lines when mapping to depth?

no idea :-)

**Reply by jumpjack** [reply]
> Map the Hue channel values to depth not the RGB to grayscale image

how?
I think I should remap the color band under the image to a 256-levels grayband... but how?

**Reply by fmw42** [reply]
The Hue channel of the RGB to HSV will be 256 grays spanning red to blue.

You could use a median filter to smooth over the black. You can then create a mask of just the black and use that to blend between the smoothed and original so that only the black gets the median smoothing

**Comment by snibgo** [comment]
I have called the input image colouredMap.png.

It has apparently been through a lossy JPEG compression, so the contour lines have halos. Yuck. Use a better source if you have one.
```
magick ^
  colouredMap.png ^
  -colorspace HCL ^
  -channel 0 -separate +channel ^
  -set colorspace sRGB ^
  -level 74.5,0%% ^
  out.png
```
![out](https://user-images.githubusercontent.com/33812368/182672542-171ade1d-97bc-43de-85b7-073de2477098.png)
Obviously, you would replace just the main map and the key. Can you see how to do that?

The contours can be made transparent, and infilled. See [Filling holes](http://im.snibgo.com/fillholes.htm) for some methods.
